### ISSUE TITLE
The meaning of providing a constant for a mutable global import

### ISSUE THREAD

**Issue by lars-t-hansen** [WebAssembly/mutable-global#9]
Suppose a module is this:
```
(module (import "m" "g" (global (mut i32))))
```
and we try to instantiate it with the import object `{m: {g: 42}}`.  Is this allowed or not?

Firefox currently allows it and there seems to be no difficulty in allowing it generally, but I think the spec is vague here.  The current spec text to cover this seems to be (document/js-api/index.bs line 335):

```
        1. If |externtype| is of the form [=𝗀𝗅𝗈𝖻𝖺𝗅=] |globaltype|,
            1. If |globaltype| [=Type=](|v|) is [=Number=],
                1. If |globaltype| is [=𝗂𝟨𝟦=], throw a {{LinkError}} exception.
                1. Let |value| be [=ToWebAssemblyValue=](|v|, |globaltype|.<em>[=global type|valtype=]</em>)
                1. Let |store| be the current agent's [=associated store=].
                1. Let (|store|, |globaladdr|) be [=alloc_global=](|store|, |globaltype|, |value|).
                1. Set the current agent's [=associated store=] to |store|.
```

A globaltype can't be just a JS type I think, it needs to carry a mutability specifier.  One could argue that since this is a number value the implicit mutability specifier is const, but need it be so?

**Comment by rossberg** [comment]
Allowing an implicit coercion from Number to mutable Global does not seem desirable.

I think the occurrence of |globaltype| in the second line of the snippet is just a typo?

**Comment by lars-t-hansen** [comment]
I agree that it makes most sense for it to be immutable (and we can always open it up later), I just wanted to make sure the decision was made explicitly.  The spec text needs to synthesize the `const` part of the globaltype here explicitly.

@rossberg, yes I was confused by that |globaltype|, couldn't tell if it was magic spec syntax or what.
